proc/utils.go

```diff
@@ -75,21 +79,30 @@ func (t *TestSource) Parents() []Proc { return nil }
 // output of the proc.  Always end a test case with Finish() to ensure
 // there weren't any unexpected records or warnings.
 type ProcTest struct {
-	ctx          Context
+	ctx          *Context
 	compiledProc Proc
 	eos          bool
 }
 
-func NewProcTest(code string, resolver *resolver.Table, inRecords []zson.Batch) (*ProcTest, error) {
-	ctx := Context{
+func NewProcTestWithProc(proc Proc, ctx *Context) *ProcTest {
```

Comment:
Since this is a proc unit testing framework, passing a proc object seems like the common case over passing zql source. So I think this should be `NewProcTest` and `NewProcTest` below should become `NewProcTestFromSource` (or some other name). 

But... author has the last word.

Reply:
I agree.